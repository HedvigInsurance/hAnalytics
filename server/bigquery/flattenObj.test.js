const flattenObj = require("./flattenObj");

test("flattens correctly", async () => {
  expect(
    flattenObj({
      property: {
        memberId: "hello",
      },
    })
  ).toMatchSnapshot();

  expect(
    flattenObj({
      property: {
        store: {
          fisk: "mock",
          other: "two",
        },
      },
    })
  ).toMatchSnapshot();

  expect(
    flattenObj({
      property: {
        store: {
          fisk: [
            {
              memberId: 123,
            },
          ],
          other: "two",
        },
      },
    })
  ).toMatchSnapshot();

  expect(
    flattenObj({
      property: {
        store: {
          fisk: [123],
          other: "two",
        },
      },
    })
  ).toMatchSnapshot();

  expect(
    flattenObj({
      property: {
        store: {
          fisk: [
            {
              memberId: [
                {
                  someProperty: "hello",
                },
              ],
            },
          ],
          other: "two",
        },
      },
    })
  ).toMatchSnapshot();

  expect(
    flattenObj({
      property: {
        initiated_from: "test",
      },
    })
  ).toMatchSnapshot();
});
