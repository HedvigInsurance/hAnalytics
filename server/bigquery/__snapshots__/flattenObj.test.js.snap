// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattens correctly 1`] = `
Object {
  "property_member_id": "hello",
}
`;

exports[`flattens correctly 2`] = `
Object {
  "property_store_fisk": "mock",
  "property_store_other": "two",
}
`;

exports[`flattens correctly 3`] = `
Object {
  "property_store_fisk": Array [
    Object {
      "member_id": 123,
    },
  ],
  "property_store_other": "two",
}
`;

exports[`flattens correctly 4`] = `
Object {
  "property_store_fisk": Array [
    123,
  ],
  "property_store_other": "two",
}
`;

exports[`flattens correctly 5`] = `
Object {
  "property_store_fisk": Array [
    Object {
      "member_id": Array [
        Object {
          "some_property": "hello",
        },
      ],
    },
  ],
  "property_store_other": "two",
}
`;
